### PR TITLE
Adjust error handling

### PR DESCRIPTION
**Pull request by bcb**
- Replace InvalidArgumentsError with InvalidParamsError,
- Remove KeyError handling.
- Inherit custom exceptions from Exception

### CODE REVIEW

File: jsonrpcserver/dispatcher.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def call(method: Method, *args: Any, **kwargs: Any) -> Any:
         The "result" part of the JSON-RPC response (the return value from the method).
 
     Raises:
-        TypeError: If arguments don't match function signature.
+        InvalidParamsError: If arguments don't match function signature.
```
Comment (steinymity) [inline jsonrpcserver/dispatcher.py]: Should we mention `ApiError` here since methods are now explicitly expected to raise them?

Comment (bcb) [reply jsonrpcserver/dispatcher.py]: The apierror should be in the method’s docstring. Really it shouldn’t have invalid params error here either, that should be in validate args. I’ll remove it.

Comment (steinymity) [reply jsonrpcserver/dispatcher.py]: Ok, agree, however, I actually prefer to document all exceptions a function could throw during "normal" operation — which includes transitive exceptions thrown by called functions. But skipping them is fine for me, too.

Comment (bcb) [reply jsonrpcserver/dispatcher.py]: That would be a huge task. Think of all the exceptions thrown by the whole tree of abstractions.

On another note, I was just thinking I’d prefer the lookup method to be a plain function, not an object method, to fit with the rest of the code.

Comment (steinymity) [reply jsonrpcserver/dispatcher.py]: Yeah, true, that's a lot of work…
Hmm, well, I actually prefer the more object oriented style with classes encapsulating data and providing access only via defined and documented methods to the more C-stylish approach keeping data and functions separate. Usually higher abstractions make it easier to use an interface correctly and harder to use it wrong. So, I'd rather stay with the `lookup()` function. This also better fits into the `Methods` class which also has an `add()` method. With both methods, direct access to `items` is not necessary anymore.

Comment (bcb) [reply jsonrpcserver/dispatcher.py]: These libraries were rewritten in the functional style after I read “Functional python programming“ (incredibly book btw I highly recommend it). I’ve never been a fan of OOP.

Comment (steinymity) [reply jsonrpcserver/dispatcher.py]: Ok, I see. Well, I usually tend to use functional-style constructs within an OOP context in methods implementation to ease implementation.
But I'm fine with a complete functional approach — so I don't mind if you change that here.

**Comment by steinymity** [comment]
Ok, so, do you still want to change anything? Otherwise, I think, we can just merge this PR.

**Comment by bcb** [comment]
Just the docstring. Should be able to make the change this morning on the way to work.

**Comment by bcb** [comment]
Ok that's done. Merge when ready.

**Comment by steinymity** [comment]
Ok, looks good to me.

**Pull request merged**